action.yml

```diff
@@ -26,6 +26,196 @@ inputs:
     description: 'GitHub token'
     required: true
     default: ${{ github.token }}
+outputs:
+  ghcr_up:
+    description: 'Was GHCR supposedly available? (true/false)'
+    value: ${{ steps.finale.outputs.ghcr_up }}
+  image_pulled_from_ghcr:
+    description: 'Was the Docker image pulled from GHCR? (true/false)'
+    value: ${{ steps.finale.outputs.image_pulled_from_ghcr }}
+  image_built_locally:
+    description: 'Was the Docker image built locally? (true/false)'
+    value: ${{ steps.finale.outputs.image_built_locally }}
+  image_ref:
+    description: 'The published URL or a unique build label for the Docker image used.'
+    value: ${{ steps.finale.outputs.image_ref }}
 runs:
-  using: 'docker'
-  image: 'docker://ghcr.io/actions/jekyll-build-pages:v1.0.7'
+  using: composite
+  steps:
+    - name: Calculate Docker tag information
+      id: action_meta
+      shell: bash
+      env:
+        DOCKER_TAG_FALLBACK: v1.0.7
+        # Do not set this as `GITHUB_ACTION_REPOSITORY` due to a bug in the runner!
+        #GITHUB_ACTION_REPO: ${{ github.action_repository }}
+        GITHUB_ACTION_REPO: actions/jekyll-build-pages
+        # Infer Docker container image to use, if possible
+        # Do not set this as `GITHUB_ACTION_REF` due to a bug in the runner!
+        GITHUB_ACTION_REF_COMMITISH: ${{ github.action_ref }}
+      run: |
+        CLEAN_COMMITISH=$(echo "${GITHUB_ACTION_REF_COMMITISH}" | sed 's/[^a-zA-Z0-9._-]//g')
+
+        # Check for the default branch
+        if [[ "${CLEAN_COMMITISH}" == "main" ]]; then
+          DOCKER_TAG="latest"
+          echo "Using Docker tag '${DOCKER_TAG}' for git default branch '${CLEAN_COMMITISH}'"
+        # Check for format of either major version tag ("v1") or a standard SemVer tag ("v1.0.7")
+        elif [[ "${CLEAN_COMMITISH}" =~ ^v[0-9]+(\.[0-9]+\.[0-9]+)?$ ]]; then
+          DOCKER_TAG="${CLEAN_COMMITISH}"
+          echo "Using Docker tag '${DOCKER_TAG}' for git version tag '${CLEAN_COMMITISH}'"
+        else
+          DOCKER_TAG="${DOCKER_TAG_FALLBACK}"
+          echo "Using Docker tag fallback '${DOCKER_TAG}' for git non-default branch or commit SHA '${CLEAN_COMMITISH}'"
+        fi
+
+        # Set step outputs
+        echo "docker_tag=${DOCKER_TAG}" >> $GITHUB_OUTPUT
+        echo "docker_repo=${GITHUB_ACTION_REPO}" >> $GITHUB_OUTPUT
+        echo "docker_registry=ghcr.io" >> $GITHUB_OUTPUT
+
+    - name: Check GHCR status
+      id: ghcr
+      shell: bash
+      run: |
+        # Using an internal endpoint would be better than the status page
+        # but this is an acceptable starting point
+        if [ -z "${GHCR_STATUS}" ]; then
+          PACKAGES_STATUS=$(curl -LSs https://www.githubstatus.com/api/v2/components.json | jq -r '.components[] | select(.name == "Packages") | .status')
```

Comment:
I think that's the limit of this design but that typically would not help us much. When Packages has a "hiccup" they don't status but it can last enough times (~ 5 minutes) for our alerting to go off 😢